static/solo/src/js/player/index.js

```diff
@@ -86,14 +87,53 @@ define(function(require) {
           player.on('tvp:media:videoended', handleEnded);
           player.on('tvp:media:ready', function(){
             playerReady = true;
+            if(player && settings && settings.hasOwnProperty("controls")){
+              $('#'+player.options.DOMContainer.id).find('.tvp-progress-bar').css('background-color', settings.controls.seekBar.progressColor);
+            }
             if ($.isFunction(callback)) {
               callback();
             }
           });
           resize();
         };
-        var defaults = require('./settings');
-        var settings = $.extend( {}, defaults, {divId:targetId });
+
+        // Check for settings is passed or not
+        if(config && config.settings){
```

Comment:
Did you tested this?
If `settings` does not exists this will brake.
One way to check for an existing property is: `if ("undefined" !== typeof config.settings)`